src/autodiff_model.jl

```diff
@@ -49,29 +49,20 @@ mutable struct ADNLPModel <: AbstractNLPModel
   c
 end
 
-function ADNLPModel(f, x0::AbstractVector; y0::AbstractVector = eltype(x0)[],
-                    lvar::AbstractVector = eltype(x0)[], uvar::AbstractVector = eltype(x0)[],
+function ADNLPModel(f, x0::AbstractVector;
+                    lvar::AbstractVector = fill(-Inf, length(x0)), uvar::AbstractVector = fill(Inf, length(x0)),
                     lcon::AbstractVector = eltype(x0)[], ucon::AbstractVector = eltype(x0)[],
-                    c = (args...)->throw(NotImplementedError("cons")),
+                    c = x -> eltype(x0)[], y0::AbstractVector = zeros(eltype(x0), length(lcon)),
                     name::String = "Generic", lin::AbstractVector{<: Integer}=Int[])
 
   nvar = length(x0)
-  length(lvar) == 0 && (lvar = -Inf*ones(nvar))
-  length(uvar) == 0 && (uvar =  Inf*ones(nvar))
   ncon = maximum([length(lcon); length(ucon); length(y0)])
 
-  A = ForwardDiff.hessian(f, x0)
-  for i = 1:ncon
-    A += ForwardDiff.hessian(x->c(x)[i], x0) * (-1)^i
-  end
   nnzh = nvar * (nvar + 1) / 2
-  nnzj = 0
+  nnzj = nvar * ncon
 
-  if ncon > 0
-    length(lcon) == 0 && (lcon = -Inf*ones(ncon))
-    length(ucon) == 0 && (ucon =  Inf*ones(ncon))
-    length(y0) == 0   && (y0 = zeros(ncon))
-    nnzj = nvar * ncon
+  if !(length(lcon) == length(ucon) == length(y0))
+    error("lcon, ucon and y0 need to be the same length")
```

Comment:
This is causing an error if the user specifies only one of those arguments. Example: https://travis-ci.org/github/JuliaSmoothOptimizers/NLPModels.jl/jobs/677749203#L271.

That's annoying because we now must specify all of them or none of them.  Alternatively, we could specify `ncon`.

Reply:
Also we should use `@lencheck`.

Reply:
Updated to accept one, two, or three arguments.